app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/MessageCardViewHolder.kt

```diff
@@ -5,53 +5,230 @@
+                            ),
+                            tint = FirefoxTheme.colors.iconPrimary
+                        )
+                    }
+                }
+                PrimaryText(
+                    text = message.data.text,
+                    modifier = Modifier.fillMaxWidth(),
+                    fontSize = 14.sp
+                )
+            } else {
+                Row(
+                    modifier = Modifier.fillMaxWidth(),
+                ) {
+                    PrimaryText(
+                        text = message.data.text,
+                        modifier = Modifier.weight(1f),
+                        fontSize = 14.sp
+                    )
+                    IconButton(
+                        modifier = Modifier.size(24.dp),
+                        onClick = onMessageClosedClicked
+                    ) {
+                        Icon(
+                            painter = painterResource(R.drawable.ic_close),
+                            contentDescription = stringResource(
+                                R.string.remove_home_collection_placeholder_content_description
+                            ),
+                            tint = FirefoxTheme.colors.iconPrimary
+                        )
+                    }
+                }
             }
-        }
+            if (!message.data.buttonLabel.isNullOrBlank()) {
+                Spacer(modifier = Modifier.height(16.dp))
 
-        binding.close.apply {
-            increaseTapArea(CLOSE_BUTTON_EXTRA_DPS)
-            setOnClickListener {
-                interactor.onMessageClosedClicked(message)
+                MessageCardButton(
+                    onClick = onMessageClicked
+                )
             }
         }
-        interactor.onMessageDisplayed(message)
     }
+}
 
-    companion object {
-        internal const val LAYOUT_ID = R.layout.nimbus_message_card
-        private const val CLOSE_BUTTON_EXTRA_DPS = 38
+/**
+ * Message Card Button.
+ *
+ * @param onClick Invoked when user clicks on button.
+ */
+
+@Suppress("LongMethod")
+@Composable
+fun MessageCardButton(onClick: () -> Unit) {
+    Button(
+        onClick = onClick,
+        modifier = Modifier
+            .clip(RoundedCornerShape(size = 4.dp))
+            .fillMaxWidth()
+            .height(36.dp),
+        elevation = ButtonDefaults.elevation(defaultElevation = 0.dp, pressedElevation = 0.dp),
+        colors = ButtonDefaults.outlinedButtonColors(backgroundColor = FirefoxTheme.colors.actionPrimary),
+    ) {
+        Text(
+            text = stringResource(R.string.preferences_set_as_default_browser),
+            modifier = Modifier.align(Alignment.CenterVertically),
+            color = FirefoxTheme.colors.textOnColorPrimary,
+            fontSize = 14.sp,
+            fontWeight = FontWeight.SemiBold,
+            letterSpacing = 0.sp,
+            maxLines = 2
+        )
+    }
```

Comment:
I am expecting that this will be refactored into its own component as discussed from your CollectionsPlaceholder work. When we do have the button as its own component, I would drop `MessageCardButton` wrapper and use the button component directly